cpp/src/arrow/util/basic_decimal.h

```diff
@@ -388,6 +399,31 @@ class ARROW_EXPORT BasicDecimal256 {
   /// \brief In-place division.
   BasicDecimal256& operator/=(const BasicDecimal256& right);
 
+  /// \brief Get the maximum decimal value (is not a valid value).
+  static inline constexpr BasicDecimal256 GetMaxSentinel() {
+#if ARROW_LITTLE_ENDIAN
+    return BasicDecimal256({std::numeric_limits<uint64_t>::max(),
+                            std::numeric_limits<uint64_t>::max(),
+                            std::numeric_limits<uint64_t>::max(),
+                            static_cast<uint64_t>(std::numeric_limits<int64_t>::max())});
+#else
+    return BasicDecimal256({static_cast<uint64_t>(std::numeric_limits<int64_t>::max()),
+                            std::numeric_limits<uint64_t>::max(),
+                            std::numeric_limits<uint64_t>::max(),
+                            std::numeric_limits<uint64_t>::max()});
+#endif
```

Comment:
Can be simplified as
`return BasicDecimal256(BitUtil::LittleEndianArray::ToNative<uint64_t, 4>({little-endian-array-0,1,2,3})`

Reply:
Or maybe `(BasicDecimal256(1) << 255) - 1`? cc @pitrou

Reply:
The former looks more desirable to me, as it can be computed at compile time.